chapters/sampling-randomization-power.tex

```diff
@@ -1,386 +1,428 @@
-If the randomization seed for the statistical software is not set,
-then its pseudorandom algorithm will pick up where it left off.
-By setting the seed, you force it to restart from the same point.
-In Stata, \texttt{set seed [seed]} will accomplish this.
-
-The code below code loads and sets up the \texttt{auto.dta} dataset
-for any random process. Note the three components: versioning, sorting, and seeding.
-Why are \texttt{check1} and \texttt{check3} the same? Why is \texttt{check2} different?
-
-\codeexample{replicability.do}{./code/replicability.do}
-
-Commands like \texttt{bys:} and \texttt{merge} will re-sort your data as part of their execution,
-To reiterate: any process that includes a random component
-is a random process, including sampling, randomization, power calculation,
-and many algorithms like bootstrapping.
-and other commands may alter the seed without you realizing it.\sidenote{\url{https://dimewiki.worldbank.org/wiki/Randomization_in_Stata}}
-Any of these things will cause the output to fail to replicate.
-Therefore, each random process should be independently executed
-to ensure that these three rules are followed.
-Before shipping the results of any random process,
+We recommend using \texttt{version 13.1} for back-compatibility;
+the algorithm was changed after Stata 14 but its improvements do not matter in practice.
+(Note that you will \textit{never} be able to transfer a randomization to another software such as R.)
+The \texttt{ieboilstart} command in \texttt{ietoolkit} provides functionality to support this requirement.\sidenote{
+  \url{https://dimewiki.worldbank.org/wiki/ieboilstart}}
+We recommend you use \texttt{ieboilstart} at the beginning of your master do-file.\sidenote{
+  \url{https://dimewiki.worldbank.org/wiki/Master_Do-files}}
+However, note that testing your do-files without running them
+via the master do-file may produce different results,
+since Stata's \texttt{version} expires after execution just like a \texttt{local}.
+
+\textbf{Sorting} means that the actual data that the random process is run on is fixed;
+because numbers are assigned to each observation in sequence,
+changing their order will change the result of the process.
+A corollary is that the underlying data must be unchanged between runs:
+you must make a fixed final copy of the data when you run a randomization for fieldwork.
+In Stata, the only way to guarantee a unique sorting order is to use
+\texttt{isid [id\_variable], sort}. (The \texttt{sort, stable} command is insufficient.)
+You can additionally use the \texttt{datasignature} commannd to make sure the data is unchanged.
+
+\textbf{Seeding} means manually setting the start-point of the randomization algorithm.
+You can draw a six-digit seed randomly by visiting \url{http://bit.ly/stata-random}.
+(This link is a shortcut to a specific request on \url{https://www.random.org}.)
```

Comment:
It is a bit unclear to me @luizaandrade , what you mean should be modified to what

Reply:
Right. This is a specific request, right? Otherwise, why have a bit.ly link instead of just telling people to go to random.org? The number created by this request has a certain number of characters and, I assumed, other desired properties?

Reply:
Not that much special: it's just six digits, ie uniform over [100,000 - 999,999]. That's all that's needed, right? Clarified in 1b9d0dd9e1ac7b7396ee5dde08ff1263565509f1

Reply:
Not that much special: it's just six digits, ie uniform over [100,000 - 999,999]. That's all that's needed, right? Clarified in 1b9d0dd9e1ac7b7396ee5dde08ff1263565509f1